python/cuspatial/cuspatial/core/binops/distance_dispatch.py

```diff
@@ -0,0 +1,190 @@
+        False,
+        (CollectionType.MULTI, CollectionType.SINGLE),
+    ),
+    (Feature_Enum.MULTIPOINT, Feature_Enum.MULTIPOINT): (
+        pairwise_point_distance,
+        False,
+        (CollectionType.MULTI, CollectionType.MULTI),
+    ),
+    (Feature_Enum.MULTIPOINT, Feature_Enum.LINESTRING): (
+        pairwise_point_linestring_distance,
+        False,
+        (CollectionType.MULTI,),
+    ),
+    (Feature_Enum.MULTIPOINT, Feature_Enum.POLYGON): (
+        pairwise_point_polygon_distance,
+        False,
+        (CollectionType.MULTI,),
+    ),
+}
+
+
+class DistanceDispatch:
+    """Dispatches distance operations between two GeoSeries"""
+
+    def __init__(self, lhs, rhs, align):
+        if align:
+            self._lhs, self._rhs = lhs.align(rhs)
+        else:
+            self._lhs, self._rhs = lhs, rhs
+
+        self._align = align
+        self._res_index = lhs.index
+        self._non_null_mask = self._lhs.notna() & self._rhs.notna()
+        self._lhs = self._lhs[self._non_null_mask]
+        self._rhs = self._rhs[self._non_null_mask]
+
+        # TODO: This test is expensive, so would be nice if we can cache it
+        self._lhs_type = self._determine_series_type(self._lhs)
+        self._rhs_type = self._determine_series_type(self._rhs)
+
+    def _determine_series_type(self, s):
+        """Check single geometry type of `s`."""
+        if contains_only_multipoints(s):
+            typ = Feature_Enum.MULTIPOINT
+        elif contains_only_points(s):
+            typ = Feature_Enum.POINT
+        elif contains_only_linestrings(s):
+            typ = Feature_Enum.LINESTRING
+        elif contains_only_polygons(s):
+            typ = Feature_Enum.POLYGON
+        else:
+            raise NotImplementedError(
+                "Geoseries with mixed geometry types are not supported"
+            )
+        return typ
+
+    def _column(self, s, typ):
+        """Get column of `s` based on `typ`."""
+        if typ == Feature_Enum.POINT:
+            return s.points.column()
+        elif typ == Feature_Enum.MULTIPOINT:
+            return s.multipoints.column()
+        elif typ == Feature_Enum.LINESTRING:
+            return s.lines.column()
+        elif typ == Feature_Enum.POLYGON:
+            return s.polygons.column()
+
+    @property
+    def _lhs_column(self):
+        return self._column(self._lhs, self._lhs_type)
+
+    @property
+    def _rhs_column(self):
+        return self._column(self._rhs, self._rhs_type)
+
+    def __call__(self):
+        func, reverse, collection_types = type_to_func[
+            (self._lhs_type, self._rhs_type)
+        ]
+        if reverse:
+            dist = func(*collection_types, self._rhs_column, self._lhs_column)
+        else:
+            dist = func(*collection_types, self._lhs_column, self._rhs_column)
+
+        # Rows with misaligned indices contains nan. Here we scatter the
+        # distance values to the correct indices.
+        res = full(
```

Comment:
What does `full` do? What does `res` mean?  If it's `result`, no need to abbreviate it.

Reply:
It's essentially an allocate and fill with specified value.

https://github.com/rapidsai/cudf/blob/abb59c83128f956c7edcb4d7744cb0faecf0026c/python/cudf/cudf/core/column/column.py#L2541